fiftyone/server/routes/sort.py

```diff
@@ -25,31 +24,22 @@ async def post(self, request: Request, data: dict):
         filters = data.get("filters", {})
         stages = data.get("view", None)
         dist_field = data.get("dist_field", None)
-        dataset = fod.load_dataset(dataset_name)
 
-        changed = False
-        if dist_field and not dataset.get_field(dist_field):
-            dataset.add_sample_field(dist_field, fof.FloatField)
-            changed = True
-
-        fosv.get_view(dataset_name, stages=stages, filters=filters)
-
-        state = fose.get_state().copy()
         view = fosv.get_view(dataset_name, stages=stages, filters=filters)
-        state.dataset = view._dataset
+        dataset = view._dataset
 
-        if isinstance(view, fov.DatasetView):
-            state.view = view
-        else:
-            view = None
+        state = fose.get_state().copy()
+        state.dataset = dataset
+        state.view = view
 
-        return {
-            "dataset": await serialize_dataset(
+        if dist_field and not dataset.has_field(dist_field):
+            dataset.add_sample_field(dist_field, fof.FloatField)
+            _dataset = await serialize_dataset(
                 dataset_name=dataset_name,
-                serialized_view=stages,
-                view_name=view.name,
+                serialized_view=view._serialize(),
+                saved_view_slug=fou.to_slug(view.name) if view.name else None,
```

Comment:
good catch on the change in param names, but won't `view.name` here always be null since the view is built from `fosv.get_view(dataset_name, stages=stages, filters=filters)`

Reply:
Yes it will always be null